fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java

```diff
@@ -114,20 +116,105 @@ public class TransactionLoadAction extends RestBaseAction {
     // Map operation name to metrics
     private final Map<TransactionOperation, OpMetrics> opMetricsMap = new HashMap<>();
 
-    private final ReadWriteLock txnNodeMapAccessLock = new ReentrantReadWriteLock();
-    private final Map<String, Long> txnNodeMap = new LinkedHashMap<>(512, 0.75f, true) {
-        @Override
-        protected boolean removeEldestEntry(Map.Entry<String, Long> eldest) {
-            return size() > (GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalBackendNumber() +
-                    GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalComputeNodeNumber()) * 512;
-        }
-    };
+    private final TransactionLoadLabelCache txnNodeMap = new TransactionLoadLabelCache();
 
     public TransactionLoadAction(ActionController controller) {
         super(controller);
         initMetrics();
     }
 
+    public static class TransactionLoadLabelCache  {
+        private final Cache<String, Long> cache;
+
+        public TransactionLoadLabelCache() {
+            Caffeine<Object, Object> builder = Caffeine.newBuilder();
+            cache = builder.initialCapacity(512).maximumWeight(getMaxCapacity())
+                    .weigher((key, value) -> 1)
+                    .expireAfterAccess((long) Config.prepared_transaction_default_timeout_second + 100, TimeUnit.SECONDS)
+                    .removalListener((key, value, cause) -> {
+                        LOG.debug("Evicted transaction label node mapping: {}->{} ({})", key, value, cause);
+                    })
+                    .recordStats()
+                    .build();
+        }
+
+        public long getMaxCapacity() {
+            long current = (long) GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalBackendNumber() +
+                    GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalComputeNodeNumber();
+            return current * 512;
+        }
+
+        public void refreshCapacity() {
+            cache.policy().eviction().ifPresent(eviction -> {
+                if (eviction.getMaximum() != getMaxCapacity()) {
+                    eviction.setMaximum(getMaxCapacity());
+                }
+            });
+        }
+
+        public void put(String key, Long value) {
+            cache.put(key, value);
+            refreshCapacity();
```

Comment:
shall refresh the capacity for a time based, instead of for every put(), although `getMaxCapacity()` is relatively lightweight, but not that trivial.